provisioning_templates/finish/preseed_default_finish.erb

```diff
@@ -26,6 +26,7 @@ oses:
 /bin/sed -i "s/.*dns-search.*/\tdns-search <%= @host.domain %>/g" /etc/network/interfaces
 /bin/sed -i "s/^<%= @host.ip %>.*/<%= @host.ip %>\t<%= @host.shortname %>.<%= @host.domain %>\t<%= @host.shortname %>/g" /etc/hosts
 /bin/echo <%= @host.shortname %> > /etc/hostname
+hostname  <%= @host.shortname %>.<%= @host.domain %>
```

Comment:
two small things:

- Please wrap the domain part into `<% if @host.domain -%>`
- use `/bin/hostname`

Reply:
actually, we also don't check for having a domain a bit above, so that part can stay as-is for now...